### DISCUSSION TITLE
Using custom linear solver

### DISCUSSION THREAD

**Discussion by RiccardoAldrighetti** [coin-or/Ipopt#464] [General]
Hi all,
I'm using IPOPT solver with Pyomo, and I would like to change the linear solver using CPLEX or Gurobi. 
I saw in the documentation that there's the option 'linear_solver' that could be set to custom. However, I don't find any other parameter in order to specify my linear solver.

Does anyone find a solution to this issue?
Thanks

**Comment by svigerske** [comment]
CPLEX or Gurobi are not the kind of linear solvers that Ipopt is looking for.

The value `custom` for option `linear_solver` is for users that implement their own [interface to a solver](https://coin-or.github.io/Ipopt/classIpopt_1_1TSymLinearSolver.html) that solves linear equation systems and want to make this available via Ipopt's C++ interface without modifying Ipopt code. There are no such interfaces for CPLEX and Gurobi, as far as I know. While their LP solvers could in principle be (mis)used to solve linear equation systems, they don't provide the inertia (information on number of negative eigenvalues) that Ipopt prefers to get in addition.

**Reply by YaIzVaNa** [reply]
Hi, 

So if I understand correctly, an interface between Gurobi and ipopt could be achieved, as long as one dedicated time to coding it? Couldn't the inertia information be obtained and sent to ipopt based on Gurobi's solver information?

**Reply by svigerske** [reply]
No, because Gurobi doesn't provide inertia in its solver information.
